templates/app/angular-default/hooks/post-create.js

```diff
@@ -1,41 +1,56 @@
 'use strict';
 
-const fs = require('fs');
+const fs = require('fs-extra');
 const path = require('path');
 const spawn = require('child_process').spawn; // eslint-disable-line security/detect-child-process
 
 exports.id = 'com.appcelerator.angular.post-create';
 
 exports.init = (logger, config, cli) => {
-	cli.on('create.post.app', (creator, next) => {
+	cli.on('create.post.app', async (creator, next) => {
 		const projectName = cli.argv.name;
 		const projectPath = path.join(cli.argv['workspace-dir'], projectName);
-		const packageJsonPath = path.join(projectPath, 'app', 'package.json');
-		if (fs.existsSync(packageJsonPath)) {
-			let packageJson = JSON.parse(fs.readFileSync(packageJsonPath).toString());
-			packageJson.name = projectName;
-			fs.writeFileSync(packageJsonPath, JSON.stringify(packageJson, null, 4));
-		}
-
+		copyGitIgnore(projectPath);
+		await updatePackageJson(projectPath, { name: projectName });
 		logger.info('Installing Angular project dependencies');
-		let errorOutput = '';
-		let npmExecutable = 'npm';
-		const spawnOptions = { cwd: path.dirname(packageJsonPath) };
-		if (process.platform === 'win32') {
-			spawnOptions.shell = true;
-			npmExecutable += '.cmd';
-		}
+		await installDependencies(projectPath);
```

Comment:
Before I setup the SDK to point to my local titanium-angular, this would error and cause an unhandled promise rejection. I guess we either need to make sure the SDK handles these gracefully or maybe try/catch these?